tools/clang/lib/AST/ASTContextHLSL.cpp

```diff
@@ -850,7 +851,9 @@ CXXMethodDecl* hlsl::CreateObjectFunctionDeclarationWithParams(
                                            parmVarDecls.size());
   }
 
-  recordDecl->addDecl(functionDecl);
+  // If this is going to be part of a template decl, don't add it to the record.
+  if (!isTemplate)
+    recordDecl->addDecl(functionDecl);
```

Comment:
So `isTemplate` describes whether the object itself is a template?  Is there a way to tell this from the recordDecl?

Reply:
I should clarify the naming. It is if the function is going to be part of a FunctionTemplateDecl. In that case we need to not add it to the record since the FunctionTemplateDecl will be added to the Record and the Method is a child of the FunctionTemplateDecl.